ElementBuilder and SimpleGui changes

aws404 opened the pull request:
This pull request has 2 main components:

- Adding the addSlot methods to SimpleGui
  - These will put the elements in the first open slot found.
- Adding some extra options and documentation to GuiElementBuilder
  - This invloved some reworking of some things such as moving CustomModelData and some new options to be placed into a CompoundTag rather than there own specific variables.
  - Added the unbreakable method
  - Added the setDamage method
  - Added methods for hiding tags by referencing the TooltipSection rather than the flag implicitly
  - Moved the name and lore to MutableText from Text so we can force the client to not render them as italic unless they are actually set to italic.
  - Added the ability to set a SkullOwner for items, this takes a game profile and a server, however this is not required (see javadoc)
  - Moved the ItemStack creation to a separate method so it can be utilised elsewhere

Patbox commented:
Hi! This pr looks great, but I'm not sure about change of parameter type (Text -> MutableText) in a builder. For backwards compatibility reason I think it should still allow just text, but check/convert to mutable within method. Also you also should update animated gui element builder to match standard one.

aws404 commented:
I thought that may be the case, I've migrated it back to Text and added a instanceof check to be safe. I also updated the animated builder (whoops!) and did some other misc cleanups and additions.

Patbox commented:
I will make few minimal changes and release 0.4.0 soon!
